scheduler/job.go

```diff
@@ -0,0 +1,69 @@
+package scheduler
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/dealbot/tasks"
+	"github.com/robfig/cron/v3"
+)
+
+// Job is a set of parameters for running a job
+type Job struct {
+	shutdown  <-chan struct{}
+	entryID   cron.EntryID
+	expireAt  time.Time
+	runtime   time.Duration
+	run       int32
+	runChan   chan<- *Job
+	runCtx    context.Context
+	runCancel context.CancelFunc
+	sched     *cron.Cron
+	task      tasks.Task
+	taskCtx   context.Context
+}
+
+// Send job to worker, wait for worker to signal it is finished.
+func (j *Job) Run() {
+	// If the schedule has expired remove job from scheduler.
+	if !j.expireAt.IsZero() && time.Now().After(j.expireAt) {
+		j.sched.Remove(j.entryID)
+		return
+	}
+
+	atomic.AddInt32(&j.run, 1)
```

Comment:
I'm surprised that atomic is necessary; I'd assume that it would be impossible for a cron job to fire twice at the same time.

Reply:
It was because the race detector complained about a race here and where this is read.  Still should not be needed, because is not read until the job is read from the channel.

Reply:
Hm, if the race detector complains when we think it shouldn't, then it smells like something is wrong. Why would we ever read and write to this field at the same time? It would be extremely surprising for the race detector to be buggy :)

Reply:
OK, fixed this by copying a subset of the job info over the channel.